src/Devices/EEPROM_TC.cpp

```diff
@@ -0,0 +1,78 @@
+
+#include "EEPROM_TC.h"
+
+void EEPROM_TC::writeMacAddress(byte mac[]) {
+  // Store MAC address in EEPROM
+  if (EEPROM.read(44) == '#') {
+    for (int i = 3; i < 6; i++) {
+      mac[i] = EEPROM.read(i + 44);
+    }
+    Serial.println(F("MAC Address found in EEPROM and read"));
+  } else {
+    Serial.println(F("No MAC Address Found in EEPROM. Generating New MAC."));
+    for (int i = 3; i < 6; i++) {
+      mac[i] = TrueRandom.randomByte();
+      EEPROM.write(i + 44, mac[i]);
+    }
+    EEPROM.write(44, '#');
+  }
+}
+
+void EEPROM_TC::writepHSetPoint(double value) {
+  if (ph_set != EepromReadDouble(PH_ADDRESS)) {
+    EepromWriteDouble(PH_ADDRESS, ph_set);
+  }
+}
+
+double EEPROM_TC::readTempSetPoint() { 
+  // Load from EEPROM
+  ph_set = EepromReadDouble(PH_ADDRESS);
+  temp_set = EepromReadDouble(TEMP_ADDRESS);
```

Comment:
This line (reading the TEMP_ADDRESS) is the only code that should be in a function named `readTempSetPoint()`. The other lines should have their own methods.